### PR TITLE
Make messages relating to teams more translatable

### PR DESCRIPTION
**Pull request by heinrich5991**
Many languages other than English have cases that make it hard or
impossible to use the same "red team" or "blue team" formulation across
multiple sentences.

This allows these languages to translate the strings separately.

This comes at the cost of some code duplication.

### CODE REVIEW

**Comment by heinrich5991** [comment]
Perhaps I should write a small script that puts the current translations into that format.

File: src/game/client/gameclient.cpp [hunk 1]
```diff
@@ -1437,7 +1481,13 @@ void CGameClient::DoEnterMessage(const char *pName, int ClientID, int Team)
 {
 	char aBuf[128], aLabel[64];
 	GetPlayerLabel(aLabel, sizeof(aLabel), ClientID, pName);
-	str_format(aBuf, sizeof(aBuf), Localize("'%s' entered and joined the %s"), aLabel, GetTeamName(Team, m_GameInfo.m_GameFlags&GAMEFLAG_TEAMS));
+	switch(GetStrTeam(Team, m_GameInfo.m_GameFlags&GAMEFLAG_TEAMS))
+	{
+	case STR_TEAM_GAME: str_format(aBuf, sizeof(aBuf), Localize("'%s' entered and joined the game"), aLabel); break;
+	case STR_TEAM_RED: str_format(aBuf, sizeof(aBuf), Localize("'%s' entered and joined the red team"), aLabel); break;
+	case STR_TEAM_BLUE: str_format(aBuf, sizeof(aBuf), Localize("'%s' entered and joined the blue team"), aLabel); break;
+	case STR_TEAM_SPECTATORS: str_format(aBuf, sizeof(aBuf), Localize("'%s' entered and joine the spectators"), aLabel); break;
```
Comment (Dune-jr) [inline src/game/client/gameclient.cpp]: joined, not joine

File: src/game/client/gameclient.cpp [hunk 2]
```diff
@@ -557,12 +566,30 @@ void CGameClient::OnMessage(int MsgId, CUnpacker *pUnpacker)
...
-				str_format(aBuf, sizeof(aBuf), Localize("You were moved to %s due to team balancing"), GetTeamName(aParaI[0], m_GameInfo.m_GameFlags&GAMEFLAG_TEAMS));
-				m_pBroadcast->DoBroadcast(aBuf);
+				{
+					const char *pMsg;
+					switch(GetStrTeam(aParaI[0], TeamPlay))
+					{
+					case STR_TEAM_GAME: pMsg = Localize("You were moved to game due to team balancing"); break;
+					case STR_TEAM_RED: pMsg = Localize("You were moved to red due to team balancing"); break;
+					case STR_TEAM_BLUE: pMsg = Localize("You were moved to blue due to team balancing"); break;
+					case STR_TEAM_SPECTATORS: pMsg = Localize("You were moved to spectators due to team balancing"); break;
```
Comment (Dune-jr) [inline src/game/client/gameclient.cpp]: There are missing `the` in all 4 strings

Comment (oy) [reply src/game/client/gameclient.cpp]: Also missing the word "team" after red/blue

File: src/game/client/gameclient.cpp [hunk 3]
```diff
@@ -1455,7 +1505,15 @@ void CGameClient::DoLeaveMessage(const char *pName, int ClientID, const char *pR
...
 	char aBuf[128];
-	str_format(aBuf, sizeof(aBuf), Localize("'%2d: %s' joined the %s"), ClientID, g_Config.m_ClShowsocial ? pName : "", GetTeamName(Team, m_GameInfo.m_GameFlags&GAMEFLAG_TEAMS));
+	char aLabel[64];
+	GetPlayerLabel(aLabel, sizeof(aLabel), ClientID, pName);
+	switch(GetStrTeam(Team, m_GameInfo.m_GameFlags&GAMEFLAG_TEAMS))
+	{
+	case STR_TEAM_GAME: str_format(aBuf, sizeof(aBuf), "'%s' joined the game", aLabel); break;
+	case STR_TEAM_RED: str_format(aBuf, sizeof(aBuf), "'%s' joined the red team", aLabel); break;
+	case STR_TEAM_BLUE: str_format(aBuf, sizeof(aBuf), "'%s' joined the blue team", aLabel); break;
+	case STR_TEAM_SPECTATORS: str_format(aBuf, sizeof(aBuf), "'%s' joined the spectators", aLabel); break;
```
Comment (oy) [inline src/game/client/gameclient.cpp]: Missing Localize for all 4 lines

**Pull request merged**